daemon/cmd/deletion_queue.go

```diff
@@ -7,67 +7,83 @@ import (
 	"context"
```

Comment:
Now that this is is more modularized, would it make sense to move this in to `daemon/cmd/cni`?

Reply:
Slightly tricky as we need to block the API server while the endpoint deletions are processed. Currently that's done by explicitly having the daemon's API handler constructor depend on `*deletionQueue`, thus ensuring that it gets started first. Not exactly ideal. Perhaps cleaner would be to have e.g. endpoint API handlers themselves block waiting for "deletion queue" to finish so it's even more explicit? I'd wait with that cleanup though.